src/threadpool_scheduler.cpp

```diff
@@ -172,6 +180,10 @@ static void thread_task_loop(threadpool_data* impl, std::size_t thread_id, task_
 	// Flag indicating if we have added a continuation to the task
 	bool added_continuation = false;
 
+    if (impl->prerun) impl->prerun();
+
+    SCOPE_EXIT({if (impl->postrun) impl->postrun();});
+
```

Comment:
This should be done in `worker_thread`, not here.

Also no exceptions are thrown by this code, so you don't need all the complicated scoping stuff: just call `prerun` before the call to `thread_task_loop` and `postrun` after.

Reply:
Hmm, but there are two early return calls inside the loop.

Reply:
Which is why I said to put the call to `prerun` and `postun` in the `worker_thread` function instead.